hoomd/md/HalfStepHook.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2009-2022 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#include "hoomd/HalfStepHook.h"
+
+namespace hoomd
+    {
+namespace md
+    {
+
+//! Trampoline for HalfStepHook classes inherited in python
+class HalfStepHookPy : public HalfStepHook
```

Comment:
```suggestion
class PYBIND11_EXPORT HalfStepHookPy : public HalfStepHook
```
Also add `PYBIND11_EXPORT` to `HalfStepHook` in `HalfStepHook.h`. This will allow plugins to define half step hooks in C++.